test/handlers/URMCompoundManager.handler.sol

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.30;
+
+// External Imports
+import {CommonBase} from "forge-std/Base.sol";
+import {StdCheats} from "forge-std/StdCheats.sol";
+import {StdUtils} from "forge-std/StdUtils.sol";
+import {console} from "forge-std/console.sol";
+
+// Internal Imports
+import {URMCompoundManager} from "src/contracts/urm/URMCompoundManager.sol";
+import {FakeProtocolContract} from "test/fakes/FakeProtocolContract.sol";
+import {ProposalState} from "src/types/GovernanceTypes.sol";
+import {RollbackSet, LibRollbackSet, RollbackProposal} from "test/helpers/RollbackSet.sol";
+
+contract URMCompoundManagerHandler is CommonBase, StdCheats, StdUtils {
+  URMCompoundManager public urm;
+
+  FakeProtocolContract[] public targets;
+  bytes4[] public selectors;
+  string public constant DESCRIPTION = "test rollback";
+
+  address public admin;
+  address public guardian;
+
+  // Use RollbackSet for tracking
+  RollbackSet internal rollbackSet;
+
+  // call summary
+  mapping(bytes32 => uint256) public calls;
+
+  modifier countCall(bytes32 key) {
+    calls[key]++;
+    _;
+  }
+
+  constructor(URMCompoundManager _urm, address _admin, address _guardian, FakeProtocolContract[] memory _targets) {
+    urm = _urm;
+    admin = _admin;
+    guardian = _guardian;
+
+    for (uint256 i = 0; i < _targets.length; i++) {
+      targets.push(_targets[i]);
+    }
+
+    selectors = [FakeProtocolContract.setFee.selector, FakeProtocolContract.setFeeGuardian.selector];
+  }
+
+  /// @notice Generate a rollback proposal with random transactions
+  /// @param _rollbackFee Fee parameter for randomization
+  /// @param _rollbackGuardian Guardian parameter for randomization
+  /// @return _targets Array of target addresses
+  /// @return _values Array of values
+  /// @return _calldatas Array of calldata
+  function _generateRollback(uint256 _rollbackFee, address _rollbackGuardian)
```

Comment:
^ to introduce a variation is targets and calldata to mirror real world scenarios